### PR TITLE
Fixed setNativeProps for web

### PR DESCRIPTION
**Pull request by jmysliv**
## Description
`setNativeProps` method was ignoring previously changed style, so before the method call, I merge the current style with the previous one to overcome this issue. Detailed description: necolas/react-native-web#1935

## Changes
- added merging current style with previous one, before `setNativeProps` call

### Before

https://user-images.githubusercontent.com/48885911/129149755-9079be66-7aee-4f66-ac37-ba68de576a10.mov

### After

https://user-images.githubusercontent.com/48885911/129149768-3c142fad-9272-4631-a666-9eb23e873034.mov

### CODE REVIEW

**Review by piaskowyk — APPROVED** [review]

File: src/reanimated2/js-reanimated/index.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export const _updatePropsJS = (updates, viewRef) => {
     );
 
     if (typeof viewRef._component.setNativeProps === 'function') {
-      viewRef._component.setNativeProps({ style: rawStyles });
+      setNativeProps(viewRef._component, rawStyles);
```
Comment (chakravarthy-mm) [inline src/reanimated2/js-reanimated/index.ts:46]: Shouldn't this be viewRef.current._component?
I was trying to use with next.js and while debugging viewRef._component is always undefined
@piaskowyk

Comment (chakravarthy-mm) [reply src/reanimated2/js-reanimated/index.ts:46]: Sorry, I thought I installed 2.2.4

**Comment by nandorojo** [comment]
I'm seeing this bug on `2.2.4` when used with SVG animations.

image

@piaskowyk Is it possible `2.2.4` doesn't include the patch from https://github.com/software-mansion/react-native-reanimated/pull/2053 that fixed animated SVGs?

If I revert back to `2.2.0`, and add this patch in my app, my SVG animations are fixed:

```patch
diff --git a/node_modules/react-native-reanimated/lib/reanimated2/js-reanimated/index.web.js b/node_modules/react-native-reanimated/lib/reanimated2/js-reanimated/index.web.js
index 76e1dc0..46aa7bf 100644
--- a/node_modules/react-native-reanimated/lib/reanimated2/js-reanimated/index.web.js
+++ b/node_modules/react-native-reanimated/lib/reanimated2/js-reanimated/index.web.js
@@ -11,7 +11,22 @@ export const _updatePropsJS = (_viewTag, _viewName, updates, viewRef) => {
             acc[index][key] = value;
             return acc;
         }, [{}, {}]);
-        viewRef.current._component.setNativeProps({ style: rawStyles });
+        if (typeof viewRef.current._component.setNativeProps === 'function') {
+          viewRef.current._component.setNativeProps({ style: rawStyles });
+        } else if (Object.keys(viewRef.current._component.props).length > 0) {
+          Object.keys(viewRef.current._component.props).forEach((key) => {
+            if (!rawStyles[key]) {
+              return;
+            }
+            const dashedKey = key.replace(/[A-Z]/g, (m) => '-' + m.toLowerCase());
+            viewRef.current._component._touchableNode.setAttribute(
+              dashedKey,
+              rawStyles[key]
+            );
+          });
+        } else {
+          console.warn('It is not possible to manipulate component');
+        }
     }
 };
 global._setGlobalConsole = (_val) => {
```

**Comment by chakrihacker** [comment]
@nandorojo with next.js, mount animations with a settimeout around 200ms. With initial render I too saw the same error

**Comment by adamhari** [comment]
This is also breaking for me on web with the same error as @nandorojo with this component which only uses View and Text:

```
const AnimatedErrorText = ({ error, style }: Props) => {
  const styles = useStyles();

  const animatedContainerStyles = useAnimatedStyle(() => ({
    height: withTiming(error ? 20 : 0, { duration: 600 }),
  }));

  const animatedTextStyles = useAnimatedStyle(() => ({
    opacity: withDelay(error ? 300 : 0, withTiming(error ? 1 : 0, { duration: 600 })),
  }));

  return (
    <Animated.View style={[styles.container, style, animatedContainerStyles]}>
      <Animated.Text style={[styles.text, animatedTextStyles]}>{error?.toString()}</Animated.Text>
    </Animated.View>
  );
};
```

**Pull request merged**